### DISCUSSION TITLE
DeviceGroup with Shelly 2.5 and detached relay to switch light group on

### DISCUSSION THREAD

**Discussion by JanickGers85** [arendst/Tasmota#17533] [Q&A] [answered]
Hello there!
I'll try to explain the scenario so maybe my request will be a bit more clear :)

I have two lamps with Tasmota firmware bonded together by device group _salotto-applique_ and they switch on/off together instantly.
They are powered by two separate Shellys (one attached to a relay of a 2.5, another to a Shelly 1) which releays are always on and those relays have a rule that call light's API to toggle them when pressing the button.
I can't modify/improve the hardware configuration.

Since sometime the HTTP calls are a bit slow I tought that I can detach the switchs from relays using SwitchMode 15, then giving the relay the same device group of the lights and that would give me instant commands.

I'm trying with a test shelly but sadly it don't works when I press the switch, but if I activate the relay of the shelly from web interface it works and if I switch on the lights from its web interface I can hear the click of the relay so it fisically works.

I'm afraid that groups can't be used with detacted relay, or better it won't works with physical switch in this case.

Anyone has tried?

Thanks!

**Comment by sfromis** [comment, marked as answer]
Details of what do are not clear. Your lamps were already powered by the relays, which are "always on", and you use the button on the relay to control the light via http requests, instead of allowing the button to perform the default Tasmota action of controlling the local relay. Is that correctly understood?

In that case, I do not see what you mean by giving them the same device group, and detaching, as the relay always on should mean that no state change happens there. But maybe it is just lack of very precise details of what you exactly do....

Anyway, for fast communication http is the worst approach, expect MQTT to be faster than http due to not needing extra roundtrips each time to create and destroy IP sessions.

**Reply by JanickGers85** [reply]
Thanks for your response!
Lamps are flashed with Tasmota and relays are on since I want to use them as smart bulbs, as they are.
Detached buttons from relays are used to control bulbs using http request.
I would like to use the more fail-safe approach as possible, that's why I used direct http calls instead of MQTT that relies on a broker.
I've done a fallback to physically switch the relay in case of bulbs not reachable.

**Reply by sfromis** [reply]
I'm still not seeing precise details of your expectations around using device groups, and what your issues were trying to make it work.

Anyway.... On a general level, if you want a relay to be always on, you could config it as `Output Hi` removing the control from Tasmota. If you then config the relay number on an unused pin, you'd have a power channel doing nothing, outside of possibly participating in a device group.

About http - I simply mentioned that you had picked to *worst* strategy for sending commands, just FYI...

**Reply by JanickGers85** [reply]
I would like to use device group putting lamps and switch in a single group so I can switch on/off the lamp using the physical switch, but without changing state of the corrisponing relay which must be stay always on.

Changing to Output Hi will make me lose the possibility to use the physical switch?

Thanks for the clarification about http, I just wanted to explain why I used it instead of MQTT

**Reply by sfromis** [reply]
No, you still have the switch input, relay or not, however you cannot have a switch in a device group, as device groups are about the actions of the device, not about your actions. And I explained how you can make a dummy relay for device group participation, if you take that direction.

**Reply by JanickGers85** [reply]
Thanks a bunch!
I'll try out shortly.

**Comment by JanickGers85** [comment]
Solved using a "virtual" relay and using it in DevGroup, thanks a lot!
